### PR TITLE
Navigation start on presentation

### PR DESCRIPTION
**Pull request by JThramer**
* Fixed potential issue where `NavigationService` was started in the `NavigationViewController` initalization.

/cc @mapbox/navigation-ios

### CODE REVIEW

**Comment by akitchen** [comment]
appearance callbacks can be called more than once. is this safely reentrant?

**Comment by frederoni** [comment]
> appearance callbacks can be called more than once

I share your concern. willAppear triggers with fast app switching, opening/closing control center and more. viewDidLoad seems more appropriate but I don't get what potential issue this fixes.

File: MapboxNavigation/NavigationViewController.swift [hunk 1]
```diff
@@ -270,12 +266,17 @@ open class NavigationViewController: UIViewController {
...
+        //start the navigation service on presentation.
+        self.navigationService.start()
+        
         view.clipsToBounds = true
     }
     
     open override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
 
+
```
Comment (frederoni) [inline MapboxNavigation/NavigationViewController.swift]: nit: drop excessive linebreak

**Comment by frederoni** [comment]
would you mind rebasing atop master?

**Pull request merged**